src/zenml/models/v2/core/action.py

```diff
@@ -0,0 +1,290 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Collection of all models concerning actions."""
+
+import copy
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    Optional,
+    TypeVar,
+)
+from uuid import UUID
+
+from pydantic import Field
+
+from zenml.constants import STR_FIELD_MAX_LENGTH
+from zenml.enums import PluginSubType
+from zenml.models.v2.base.base import BaseZenModel
+from zenml.models.v2.base.page import Page
+from zenml.models.v2.base.scoped import (
+    WorkspaceScopedFilter,
+    WorkspaceScopedRequest,
+    WorkspaceScopedResponse,
+    WorkspaceScopedResponseBody,
+    WorkspaceScopedResponseMetadata,
+    WorkspaceScopedResponseResources,
+)
+from zenml.models.v2.core.trigger import TriggerResponse
+from zenml.models.v2.core.user import UserResponse
+
+if TYPE_CHECKING:
+    from zenml.zen_stores.schemas import BaseSchema
+
+    AnySchema = TypeVar("AnySchema", bound=BaseSchema)
+
+
+# ------------------ Request Model ------------------
+
+
+class ActionRequest(WorkspaceScopedRequest):
+    """Model for creating a new action."""
+
+    name: str = Field(
+        title="The name of the action.", max_length=STR_FIELD_MAX_LENGTH
+    )
+    description: str = Field(
+        default="",
+        title="The description of the action",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    flavor: str = Field(
+        title="The flavor of the action.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    plugin_subtype: PluginSubType = Field(
+        title="The subtype of the action.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    configuration: Dict[str, Any] = Field(
+        title="The configuration for the action.",
+    )
+    service_account_id: UUID = Field(
+        title="The service account that is used to execute the action.",
+    )
+    auth_window: Optional[int] = Field(
```

Comment:
We talked about this and decided that the service account should be as close to the action as possible, because the action implementation is the component that actually uses it. The trigger should be lightweight because it keeps the concerns of the event hub limited to relaying and filtering events.

Reply:
Alright, I will defer this to you then